test/integration/targets/nios_smoke_tests/tasks/nios_host_records.yml

```diff
@@ -0,0 +1,85 @@
+- name: cleanup the parent object
+  nios_zone:
+    name: ansible.com
+    state: absent
+    provider: "{{ nios_provider }}"
+
+- name: create the parent object
+  nios_zone:
+    name: ansible.com
+    state: present
+    provider: "{{ nios_provider }}"
+
+- name: cleanup ipv4 host record
+  nios_host_record:
+    name: host.ansible.com
+    ipv4:
+      - address: 192.168.10.1
+    state: absent
+    provider: "{{ nios_provider }}"
+
+- name: create an ipv4 host record
+  nios_host_record:
+    name: host.ansible.com
+    ipv4:
+      - address: 192.168.10.1
+    state: present
+    provider: "{{ nios_provider }}"
+  register: ipv4_create1
+
+- name: recreate an ipv4 host record
+  nios_host_record:
+    name: host.ansible.com
+    ipv4:
+      - address: 192.168.10.1
+    state: present
+    provider: "{{ nios_provider }}"
+  register: ipv4_create2
+
```

Comment:
@jctanner can you also include ipv6 scenarios, as bug to to support ipv6 address is completed and merged